CHANGELOG.md

```diff
@@ -1,5 +1,60 @@
+- Toggle OSI clusters with 3.1.17 for autoscale
+- Added permissions to nodes/proxy Kubernetes resource for Microsoft.Security/pricings/microsoft-defender-operator role
+- The service mesh add-on now skips validation of its compatibility with cluster version unless mesh upgrade or cluster upgrade is attempted.
+- Using [Alias Minor Version](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#alias-minor-version) is now allowed on preview versions. 
+
+
+* Bug Fixes:
+- Fixes a bug where put agentpool without a specified version in lts clusters would have an internal error.
+- Skips verification of the CNI files installed by the Azure CNS daemonset. This change is made to avoid a race condition observed where components like LinkerD and Istio doing CNI chaining mutate the CNI conflict after its written, causing the validation that the file was written correctly to fail.
+- Error message improved  to specify that it is only allowed to update public SSH key in preview API versions.
+
+* Component Updates: 
+- Upgraded Azure file CSI driver to [v1.29.4](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.29.4)  on AKS 1.28, [v1.30.1](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.30.1)  on AKS 1.29
+- Upgraded Azure Blob CSI driver to [v1.23.4](https://github.com/kubernetes-sigs/blob-csi-driver/releases/tag/v1.23.4)  on AKS 1.28 and 1.29
+- Upgraded extension manager to [v1.15.3](https://github.com/azure-core/ClusterConfigurationAgent/releases/tag/v1.15.3)
+- Azure Monitor Metrics April release to [v.6.8.7](https://github.com/Azure/prometheus-collector/blob/main/RELEASENOTES.md#release-04-08-2024)
+- Istio revision asm-1-21 is now available with Istio-based service mesh add-on. More information on performing canary upgrade for the new minor revision of Istio can be found [here](https://learn.microsoft.com/en-us/azure/aks/istio-upgrade)
+- Bump Windows containerd v1.7 to v1.7.14 in k8s v1.28+
+- Bump Windows image versions to 2024.03B 
+   - Cached new image version for csi, cni, containerinsights, cloud-node-manager.
+   - Security Intelligence Update for Microsoft Defender Antivirus is upgraded from 1.407.466.0 to 1.407.870.0.
+- Windows Server 2019 Image has been updated to [Windows Server 2019-17763.5696.240410](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2019-containerd/17763.5696.240410.txt)   
+- Windows Server 2022 has been updated to [Windows Server Gen2 20348.2340.240401 ](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2022-containerd-gen2/20348.2340.240401.txt), [Windows Server Gen1 20348.2340.240401](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2022-containerd/20348.2340.240401.txt)
```

Comment:
we need to copy these files to this repo and update the links to point to those files